src/weppy.ts

```diff
@@ -0,0 +1,478 @@
+/// <reference path="../weppy.d.ts"/>
+
+// these seem not to be defined in window
+interface Window {
+	XMLHttpRequest?: XMLHttpRequest;
+	XDomainRequest?: XMLHttpRequest;
+	JSON?: JSON;
+}
+
+module WeppyImpl {
+
+	var protocolVersion = 3,
+		pathDelimiter = '.',
+		namespaceDelimiter = '::',
+		active = false,
+		options: WeppySettings = {
+			"host": '/weppy',
+			"transport": 'url', // 'url' or 'post'
+			"active": true,
+			"sample": 0.01,
+			"aggregationInterval": 1000,
+			"maxInterval": 5000,
+			"decimalPrecision": 3,
+			"page": 'index',
+			"context": {},
+			"debug": false
+		},
+		initTime = +(new Date),
+		queue, aggregationTimeout, maxTimeout, sentPerformanceData,
+		now = () => {
+			return window.performance && window.performance.now ? window.performance.now() : +(new Date);
+		},
+		log: any = () => {
+			if (options.debug) {
+				if (typeof options.debug == 'function') {
+					options.debug.apply(window, arguments);
+				} else {
+					window.console && window.console.log && window.console.log.apply
+						? window.console.log.apply(window.console, arguments) : void 0;
+				}
+			}
+		},
+		logError: any = () => {
+			window.console && window.console.error && window.console.error.apply &&
```

Comment:
I think it doesn't hurt us and it increases open-source value for that project to check for console.log existence and according to https://developer.mozilla.org/en-US/docs/Web/API/Console.log it's a non-standard feature. I'd leave as it is.